### PR TITLE
Basic Linting of the repo

### LINKED ISSUE #4: Tidy the code (basic)
At the moment the code is rather free-form and has various unused im[ports/dead lines etc. so could do with some pruning/tidying.

I suggest applying black+pylint or ruff as a first pass to clean things up.

These would ideally then become part of a continuous integration process for the code.

### PR DESCRIPTION
**Pull request by surbhigoel77**
Fixes #4 

The PR is to lint the entire repo to improve the quality of the code. Things formatted are in the form of:
- [x] Docstrings 
- [x] Blank Lines
- [x] White spaces
- [x] Extra long lines

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation

## Key checklist

- [ ] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [ ] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by surbhigoel77** [comment]
Using `ruff` to lint the existing code. One of the most highlighted formatting error is `missing docstrings` for defined classes and functions.

**Comment by surbhigoel77** [comment]
Doctsrings have been added to all the classes and existing functions. Restructuring the code may require updating these docstrings accordingly in future.

File: pyproject.toml
Comment (jatkinson1000) [inline pyproject.toml:1]: We can perhaps have a discussion about what other rules we might want to add at the internal meeting.

**Review by jatkinson1000 — COMMENTED** [review]
All seem sensible so far - mostly formatting changes.

The docstrings are somewhat hybrid at the moment, but we should make sure they conform to the numpy style: https://numpydoc.readthedocs.io/en/latest/format.html
Main one I can spot at the moment is `args` should be `Parameters`.

**Comment by mondus** [comment]
To merge after addition of CI and once Jacks final comments is addressed (internally). Open a new issue on Docstrings so that these can be updated with a bit more detail after #5.

**Comment by surbhigoel77** [comment]
> All seem sensible so far - mostly formatting changes.
> 
> The docstrings are somewhat hybrid at the moment, but we should make sure they conform to the numpy style: https://numpydoc.readthedocs.io/en/latest/format.html Main one I can spot at the moment is `args` should be `Parameters`.


Not changing this at the moment as the ML code is being refactored. Once that is done, I will update the docstrings acc to the numpy format.

**Review by jatkinson1000 — APPROVED** [review]
I realise that this is failing CI, but that's sort of the point of this PR.
I suggest merging now and then working to come towards compliance with the rest of the code, perhaps in #5 and #6 which will require a rebase on top of main once this is merged.

**Pull request merged**